CorsixTH/Lua/dialogs/resizables/main_menu.lua

```diff
@@ -47,7 +47,12 @@ function UIMainMenu:UIMainMenu(ui)
     {_S.main_menu.exit,            self.buttonExit,           _S.tooltip.main_menu.exit}
   }
   self.no_menu_entries = #menu_items
-  self:UIResizable(ui, 200, (menu_item_height + 10) * (#menu_items + 1), col_bg)
+
+  -- Add a bit more space to menu if updates are disabled
+  local height_modifier = ui.app.config.check_for_updates and 1 or 1.25
+  -- Work out the menu's height, giving extra space for the version information
+  self.menu_items_height = math.floor((menu_item_height + 10) * (#menu_items + height_modifier))
```

Comment:
Your modifier looks like magic.

The usual approach to obtain a height (or width) for a window is to sum the heights(widths) of everything inside the window, including any spacing.
That is what `(menu_item_height + 10) * (#menu_items + 1)` said to me at least (ie `height of a single button` * `number of buttons`).

The "+ 1" in the latter is a bit weird, but that may be added space at the very top and the bottom of the window. Alternatively, someone was lazy and thought that the version numbers were mostly as high as a button so " +1" button would do the trick (instead of the better " + height_of_version_number_lines" after the last closing parenthesis).

The multiplication factor that you added has the wrong intent. In the above meaning, you stated that the total button height increased, so the buttons should become bigger or spacing between them should become larger accordingly.  
That doesn't happen though, the button size computation is off now, it allocates more height for itself than it needs, and you use that created empty space to add a new line.

The fix here is to add height for the 3 text lines, and probably also reduce some of the button height, as the spacing below the bottom button is now less high than it originally was (you'd have to check the history of the "+1" though if you reallt want to know).

Reply:
Quite an old commit by lego in an attempt to make it less magic by the looks of it:
https://github.com/CorsixTH/CorsixTH/commit/7bbe037373418749106731b7db5d7dde99e6d480

Are we trying to say a line height here is 15px? As that could be correct for the built in font, but not freetype.
There's also no logic written to get line height that I can see, so usually all our efforts in the ui are best guess.

We could go by a line being 15px just for the sake of setting the height. Window stuff isn't very friendly.
You could also just do a +15 at the end of the menu_items_height when the extra line is needed.

Nothing really screams out a good solution here.

>  also reduce some of the button height, as the spacing below the bottom button is now less high than it originally was

Not sure that's something have a real effect here.

Reply:
We have hard-coded sizes all over the place in gui code, not much harm to add a few more imho.

If you want to go on a crusade against that, you'd better engage in a full scale war by building a new way to specify window content.

Reply:
(Where we specify what it should show and the computer does the size and position computations.)